Buddies.Web/src/pages/register.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import Box from '@mui/material/Box';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import axios from 'axios';
+import RegisterForm from '../components/RegisterForm';
+import { RegisterRequest } from '../api/model/registerRequest';
+import { ValidationProblemDetails } from '../api/model/validationProblemDetails';
+
+/**
+ * Registration page.
+ */
+const Register: React.VFC = () => {
+  const formMethods = useForm<RegisterRequest>({
+    defaultValues: {
+      firstName: '',
+      lastName: '',
+      password: '',
+      email: '',
+    },
+  });
+
+  const onSubmit: SubmitHandler<RegisterRequest> = async (data) => {
+    try {
+      await axios.post('/api/v1/users/register', data);
+    } catch (error) {
+      if (axios.isAxiosError(error) && error.response) {
+        const problems = error.response as ValidationProblemDetails;
+        if (problems.errors) {
+          Object.keys(problems.errors).forEach((errorField) => {
+            if (errorField === 'firstName'
+                || errorField === 'lastName'
+                || errorField === 'email'
+                || errorField === 'password') {
+              const errorMsg = problems.errors![errorField].join('\n');
+              formMethods.setError(errorField, { message: errorMsg });
+            }
+          });
+        }
+      }
+    }
+  };
```

Comment:
I recommend having an api class which handles all of the HTTP logic. That way it's centralized and the core logic is easier to test. Furthermore, the logic for posting and getting data is 90% the same between requests - only thing that changes is the parameters, path and response. Using generics you can simplify a lot of the code.

Reply:
As discussed during weekly meeting, actual API call has been refactored into separate file for maintainability but error handling code is kept in component as it relies on local state (e.g. `setError`).